File: 2753-minimum-number-of-operations-to-make-all-array-elements-equal-to-1/2753-minimum-number-of-operations-to-make-all-array-elements-equal-to-1.cpp
class Solution {
   public:
    int n;
    int gcd(int a, int b) {
        if (b == 0) return a;
        return gcd(b, a % b);
    }
    int minOperations(vector<int>& nums) {
        n = nums.size();
        int ans = INT_MAX;

        for (int i = 0; i < n; i++) {
            if (nums[i] == 1) {
                return n - count(nums.begin(), nums.end(), 1);
            }
            int g = nums[i];
            for (int j = i + 1; j < n; j++) {
                g = gcd(g, nums[j]);
                if (g == 1) {
                    ans = min(ans, j - i + n - 1);
                    break;
                }
            }
        }

        return ans == INT_MAX ? -1 : ans;
    }
};

File: 2753-minimum-number-of-operations-to-make-all-array-elements-equal-to-1/2753-minimum-number-of-operations-to-make-all-array-elements-equal-to-1.rs
use std::cmp;
impl Solution {
    pub fn min_operations(nums: Vec<i32>) -> i32 {
        let mut ans = i32::MAX;
        let n = nums.len();

        for i in 0..n {
            if nums[i] == 1 {
                let cnt1 = nums.iter().filter(|&n| *n == 1).count();
                return (n - cnt1) as i32;
            }
            let mut g = nums[i];
            for j in i + 1..n {
                g = Self::gcd(g, nums[j]);
                if g == 1 {
                    ans = cmp::min(ans, (j - i + n - 1) as i32);
                    break;
                }
            }
        }

        ans = if ans == i32::MAX { -1 } else { ans };
        ans
    }

    pub fn gcd(a: i32, b: i32) -> i32 {
        let res = if b == 0 { a } else { Self::gcd(b, a % b) };
        res
    }
}